mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -368,11 +369,14 @@ impl S3CrtClientInner {
                 CredentialsProvider::new_anonymous(&allocator).map_err(NewClientError::ProviderFailure)?
             }
             S3ClientAuthConfig::Profile(profile_name) => {
-                let credentials_profile_options = CredentialsProviderProfileOptions {
+                let credentials_chain_default_options = CredentialsProviderChainDefaultOptions {
                     bootstrap: &mut client_bootstrap,
-                    profile_name_override: &profile_name,
+                    profile_name_override: Some(&profile_name),
                 };
-                CredentialsProvider::new_profile(&allocator, credentials_profile_options)
+                // If profile doesn't work out (e.g. does not exist), other methods will be attempted:
+                // https://github.com/awslabs/aws-c-auth/blob/9d904153eb187e5a553258de2433f740cd46813e/include/aws/auth/credentials.h#L1250
+                // Credentials will be cached for 15 minutes.
+                CredentialsProvider::new_chain_default(&allocator, credentials_chain_default_options)
                     .map_err(NewClientError::ProviderFailure)?
             }
```

Comment:
I'd recommend updating the `Default` only, and add a profile name to the override (like below).

https://github.com/dannycjones/mountpoint-s3/blob/16cb2ed1baf218083aa48e96c1059c53e6983680/mountpoint-s3-client/src/s3_crt_client.rs#L261-L272